src/data/set/disjointed.lean

```diff
@@ -57,6 +57,13 @@ theorem pairwise_disjoint_on_nat [semilattice_inf_bot α] (f : ℕ → α) :
   pairwise (disjoint on f) ↔ ∀ (m n) (h : m < n), disjoint (f m) (f n) :=
 pairwise_on_nat disjoint.symm f
 
+lemma pairwise_disjoint_on_inter_nat {f : ℕ → set α} {a : set α}
+  (hf : pairwise (disjoint on f)) : pairwise (disjoint on (λ n, a ∩ f n)) :=
+begin
+  rintro i j hij x ⟨⟨-, hx₁⟩, -, hx₂⟩,
+  exact hf i j hij ⟨hx₁, hx₂⟩,
+end
```

Comment:
This lemma doesn't require `nat` at all

Reply:
A shorter proof is
```suggestion
hf.mono $ λ i j, disjoint.mono inf_le_right inf_le_right
```

Reply:
If you wanted you could add these lemmas in another file to match `disjoint.sup_right` and `disjoint.union_left`:
```lean
lemma disjoint.inf_left {α} [semilattice_inf_bot α] {a b : α} (c : α) (h : disjoint a b) :
  disjoint (a ⊓ c) b :=
h.mono_left inf_le_left

lemma disjoint.inf_left' {α} [semilattice_inf_bot α] {a b : α} (c : α) (h : disjoint a b) :
  disjoint (c ⊓ a) b :=
h.mono_left inf_le_right

lemma disjoint.inf_right {α} [semilattice_inf_bot α] {a b : α} (c : α) (h : disjoint a b) :
  disjoint a (b ⊓ c) :=
h.mono_right inf_le_left

lemma disjoint.inf_right' {α} [semilattice_inf_bot α] {a b : α} (c : α) (h : disjoint a b) :
  disjoint a (c ⊓ b) :=
h.mono_right inf_le_right

lemma disjoint.inter_left {α} {a b : set α} (c : set α) (h : disjoint a b) :
  disjoint (a ∩ c) b :=
disjoint.inf_left _ h

lemma disjoint.inter_left' {α} {a b : set α} (c : set α) (h : disjoint a b) :
  disjoint (c ∩ a) b :=
disjoint.inf_left' _ h

lemma disjoint.inter_right {α} {a b : set α} (c : set α) (h : disjoint a b) :
  disjoint a (b ∩ c) :=
disjoint.inf_right _ h

lemma disjoint.inter_right' {α} {a b : set α} (c : set α) (h : disjoint a b) :
  disjoint a (c ∩ b) :=
disjoint.inf_right' _ h
```

Reply:
@JasonKYi  in case you missed my follow up comments.

Reply:
Where do you think these lemmas should go?

Reply:
I would think in `order.bounded_lattice`, is that correct?

Reply:
Yes, but the set versions go in `data.set.lattice`. They should go right after their `sup` / `union` counterparts.

Reply:
Should I make a new PR for this?

Reply:
Probably a good idea.

Reply:
#8407